From 378f4b4b44647a1bf708b98a471997770cf11766 Mon Sep 17 00:00:00 2001
From: Joerg Bornemann <joerg.bornemann@qt.io>
Date: Tue, 2 Jan 2024 21:11:59 +0100
Subject: [PATCH] CMake: Fix user_facing_tool_links.txt for Apple platforms

User-facing apps that are app bundles on macOS weren't linked correctly.
For example, for Qt Designer we're creating now an entry like the
following:
    .../Designer.app/Contents/MacOS/Designer /usr/bin/designer6

Fixes: QTBUG-120487
Change-Id: I3eada15c4c5bf31b9b08829333d3c87f76ede7a2
---
 cmake/QtPostProcessHelpers.cmake | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/cmake/QtPostProcessHelpers.cmake b/cmake/QtPostProcessHelpers.cmake
index 531016aba2..b9a8d0d889 100644
--- a/qtbase/cmake/QtPostProcessHelpers.cmake
+++ b/qtbase/cmake/QtPostProcessHelpers.cmake
@@ -842,8 +842,15 @@ function(qt_internal_generate_user_facing_tools_info)
         if(NOT filename)
             set(filename ${target})
         endif()
+        set(linkname ${filename})
+        if(APPLE)
+            get_target_property(is_macos_bundle ${target} MACOSX_BUNDLE )
+            if(is_macos_bundle)
+                set(filename "${filename}.app/Contents/MacOS/${filename}")
+            endif()
+        endif()
         qt_path_join(tool_target_path "${CMAKE_INSTALL_PREFIX}" "${INSTALL_BINDIR}" "${filename}")
-        qt_path_join(tool_link_path "${INSTALL_PUBLICBINDIR}" "${filename}${PROJECT_VERSION_MAJOR}")
+        qt_path_join(tool_link_path "${INSTALL_PUBLICBINDIR}" "${linkname}${PROJECT_VERSION_MAJOR}")
         list(APPEND lines "${tool_target_path} ${tool_link_path}")
     endforeach()
     string(REPLACE ";" "\n" content "${lines}")
-- 
2.34.1

